<a href="https://colab.research.google.com/github/gxilly/BasicProg_DataViz65/blob/main/Pandas_Dataframe.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

https://datavizcatalogue.com

In [2]:
import pandas as pd

## Meta Data = Data that describes data
## CSV = Comma Separated value


# นำ Data เข้า
- upload to colab (ง่ายกว่า)
- link with google drive (ดีกว่า ใช้อันนี้!)


## upload to colab 
- เวลาเปิดไฟล์ใหม่ ต้องกลับมาเอาข้อมูลใส่ไฟล์ใหม่ทุกครั้ง


In [8]:
data0 = pd.read_csv('/content/opendata_project.csv')

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4,7,26,27,28,29,30) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [9]:
data0.head()

row_number                project_id                             name_en  \
0      7880.0  5ee08d64dda791001299b2a6  Winning Residence Sukhumvit Praksa   
1     16689.0             project_11326                    Gift Condominium   
2       290.0  62739c4089233e001362b001    Sanrak 9 Sukprayoon Nong Krathum   
3     20719.0              project_2010                     Baan Sansabai 1   
4       199.0             project_16838          Baanpruksa Ratchapruek-345   

                              name_th propertytype_id propertytype_name_en  \
0  วินนิ่ง เรสซิเดนท์ สุขุมวิท แพรกษา               1       Detached House   
1                 กิ๊ฟท์ คอนโดมิเนียม               2                Condo   
2      แสนรัก 9 ศุขประยูร หนองกระทุ่ม               3             Townhome   
3                       บ้านแสนสบาย 1               1       Detached House   
4              บ้านพฤกษา ราชพฤกษ์-345               3             Townhome   

  propertytype_name_th  price_min     developer_id  \
0                 บ้าน  3790000.0  developer_10678   
1                คอนโด  1100000.0   developer_7319   
2             ทาวน์โฮม        0.0  developer_10510   
3                 บ้าน        0.0  developer_10510   
4             ทาวน์โฮม  1930000.0   developer_1026   

              developer_name_en  ... facility_meeting  facility_park  \
0  Winning Asset Plus Co., Ltd.  ...              NaN            NaN   
1      Gift Asset Land Co.,Ltd.  ...              NaN            1.0   
2                           NaN  ...              NaN            NaN   
3                           NaN  ...              NaN            NaN   
4  Preuksa Real Estate Co.,Ltd.  ...              NaN            1.0   

   facility_playground facility_pool facility_security         date_created  \
0                  NaN           1.0               1.0  2020-06-10 14:36:04   
1                  NaN           1.0               1.0  2017-10-30 15:38:48   
2                  NaN           NaN               NaN  2022-05-05 16:43:28   
3                  NaN           NaN               NaN  2016-09-05 15:32:15   
4                  NaN           1.0               1.0  2019-09-05 11:49:37   

           date_finish         date_updated  source  \
0                  NaN  2020-12-04 10:32:58  baania   
1  2013-01-01 07:00:00  2020-12-04 10:49:00  baania   
2                  NaN  2022-05-05 17:30:03  baania   
3  1970-01-02 07:00:00  2020-12-04 10:45:31  baania   
4  1970-01-02 07:00:00  2021-03-17 12:00:04  baania   

                                         url_project  
0  https://baania.com/th/project/วินนิ่ง-เรสซิเดน...  
1  https://baania.com/th/project/กิ๊ฟท์-คอนโดมิเน...  
2  https://baania.com/th/project/แสนรัก-9-ศุขประย...  
3  https://baania.com/th/project/บ้านแสนสบาย-1-pr...  
4  https://baania.com/th/project/บ้านพฤกษา-ราชพฤก...  

[5 rows x 45 columns]

## link with google drive

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [6]:
data1 = pd.read_csv('/content/drive/MyDrive/BPDV65/opendata_unittype.csv')
data1.head()

row_number     unittype_id    project_id             name_th  \
0       27501   unittype_9565  project_2446           2 Bedroom   
1       28777  unittype_10267  project_3543  บ้านแฝดสองชั้น Z-2   
2       28865    unittype_471   project_143    1 Bedroom Type B   
3       24988  unittype_19506  project_7766    Studio Floor 1-2   
4       38596  unittype_10290  project_3551                TH 3   

   propertytype_id propertytype_name_en propertytype_name_th  \
0                2                Condo                คอนโด   
1            20000           Twin house              บ้านแฝด   
2                2                Condo                คอนโด   
3                2                Condo                คอนโด   
4                3             Townhome             ทาวน์โฮม   

   area_total_min_wa  area_usable_min  price_min  ...  count_room_guest  \
0                NaN             37.3  2500000.0  ...               1.0   
1               40.0            150.0  2390000.0  ...               NaN   
2                NaN             28.0        NaN  ...               NaN   
3                NaN             27.0   170000.0  ...               NaN   
4                NaN            182.0  2390000.0  ...               1.0   

   count_room_kitchen  count_room_living  count_room_maid  count_room_storage  \
0                 1.0                NaN              NaN                 NaN   
1                 NaN                NaN              NaN                 NaN   
2                 NaN                NaN              NaN                 NaN   
3                 NaN                NaN              NaN                 NaN   
4                 1.0                NaN              NaN                 2.0   

   count_room_utility         date_created         date_updated  source  \
0                 NaN  2021-11-19 18:00:09  2021-11-19 17:44:45  baania   
1                 NaN  2022-01-27 14:30:04  2022-01-27 14:22:55  baania   
2                 NaN  2022-01-14 15:00:10  2022-01-14 14:40:04  baania   
3                 NaN  2020-12-04 10:40:59  2020-10-14 01:06:48  baania   
4                 NaN  2021-11-12 13:30:08  2021-02-03 12:12:54  baania   

                                        url_unittype  
0  https://baania.com/th/project/เฟ-ลิค-คอนโด-ลาด...  
1  https://baania.com/th/project/เซน-บาย-ศิวาลัย-...  
2  https://baania.com/th/project/วันพลัส-นายน์ทีน...  
3  https://baania.com/th/project/บ้านสวนชมน้ำ-776...  
4  https://baania.com/th/project/เบล็สทาวน์-ศรีนค...  

[5 rows x 27 columns]

In [ ]:
data1.head(10) # ดูหัวตาราง

row_number               unittype_id                project_id  \
0       27501             unittype_9565              project_2446   
1       28777            unittype_10267              project_3543   
2       28865              unittype_471               project_143   
3       24988            unittype_19506              project_7766   
4       38596            unittype_10290              project_3551   
5       30766             unittype_8026              project_2694   
6       33193            unittype_23233              project_7531   
7        1666            unittype_14501              project_5799   
8       41823  6215b4a7a441d30014e1f02f  6215b1cfc1620300140bc087   
9       34458            unittype_34564             project_14884   

                      name_th  propertytype_id propertytype_name_en  \
0                   2 Bedroom                2                Condo   
1          บ้านแฝดสองชั้น Z-2            20000           Twin house   
2            1 Bedroom Type B                2                Condo   
3            Studio Floor 1-2                2                Condo   
4                        TH 3                3             Townhome   
5               Luciano Grand                1       Detached House   
6         อาคารพาณิชย์สามชั้น                4           Commercial   
7    บ้านเดี่ยวสองชั้น Type A                1       Detached House   
8  บ้านเดี่ยวชั้นเดียว Type C                1       Detached House   
9         บ้านเดี่ยวชั้นเดียว                1       Detached House   

  propertytype_name_th  area_total_min_wa  area_usable_min   price_min  ...  \
0                คอนโด                NaN             37.3   2500000.0  ...   
1              บ้านแฝด               40.0            150.0   2390000.0  ...   
2                คอนโด                NaN             28.0         NaN  ...   
3                คอนโด                NaN             27.0    170000.0  ...   
4             ทาวน์โฮม                NaN            182.0   2390000.0  ...   
5                 บ้าน               70.5            198.0  11700000.0  ...   
6         อาคารพาณิชย์                NaN            180.0   2990000.0  ...   
7                 บ้าน                NaN              NaN   3300000.0  ...   
8                 บ้าน                NaN             80.0    990000.0  ...   
9                 บ้าน                NaN              NaN         NaN  ...   

   count_room_guest  count_room_kitchen  count_room_living  count_room_maid  \
0               1.0                 1.0                NaN              NaN   
1               NaN                 NaN                NaN              NaN   
2               NaN                 NaN                NaN              NaN   
3               NaN                 NaN                NaN              NaN   
4               1.0                 1.0                NaN              NaN   
5               NaN                 NaN                NaN              NaN   
6               NaN                 NaN                NaN              NaN   
7               NaN                 NaN                NaN              NaN   
8               NaN                 NaN                NaN              NaN   
9               NaN                 NaN                NaN              NaN   

   count_room_storage  count_room_utility         date_created  \
0                 NaN                 NaN  2021-11-19 18:00:09   
1                 NaN                 NaN  2022-01-27 14:30:04   
2                 NaN                 NaN  2022-01-14 15:00:10   
3                 NaN                 NaN  2020-12-04 10:40:59   
4                 2.0                 NaN  2021-11-12 13:30:08   
5                 NaN                 NaN  2021-11-12 13:22:43   
6                 NaN                 NaN  2020-12-04 10:46:42   
7                 NaN                 NaN  2020-12-04 10:36:59   
8                 NaN                 NaN  2022-02-23 12:00:12   
9                 NaN                 NaN  2020-12-04 10:45:1

In [ ]:
data1.tail(10) # ดูท้ายตาราง จากล่างขึ้นมา

row_number               unittype_id                project_id  \
42615       36172            unittype_34140             project_14663   
42616       28981            unittype_25373             project_10133   
42617       35109             unittype_1459               project_711   
42618       22299            unittype_13674              project_5356   
42619       11944            unittype_34884             project_15005   
42620       37109            unittype_38508             project_16864   
42621          14            unittype_14016              project_5275   
42622       41009  617bde1687dd2300149d2195  617bdd47f62011001350de07   
42623        9333            unittype_37375             project_16389   
42624         780            unittype_17011              project_6762   

                   name_th  propertytype_id propertytype_name_en  \
42615            2 Bedroom                2                Condo   
42616    บ้านเดี่ยวสองชั้น                1       Detached House   
42617          บ้านแสงรุ่ง                3             Townhome   
42618  บ้านเดี่ยวชั้นเดียว                1       Detached House   
42619             Ametrine                1       Detached House   
42620               Canola                1       Detached House   
42621             Grandeur                1       Detached House   
42622               Grande                3             Townhome   
42623  บ้านเดี่ยวชั้นเดียว                1       Detached House   
42624             บ้านธารา                1       Detached House   

      propertytype_name_th  area_total_min_wa  area_usable_min   price_min  \
42615                คอนโด                NaN             60.0   3500000.0   
42616                 บ้าน                NaN            200.0         NaN   
42617             ทาวน์โฮม                NaN              NaN         NaN   
42618                 บ้าน                NaN            130.0  36500000.0   
42619                 บ้าน                NaN            353.0  23590000.0   
42620                 บ้าน                NaN            157.0         NaN   
42621                 บ้าน                NaN            223.0   7490000.0   
42622             ทาวน์โฮม                NaN            133.0         NaN   
42623                 บ้าน                NaN            180.0   3200000.0   
42624                 บ้าน                NaN            170.0   3790000.0   

       ...  count_room_guest  count_room_kitchen  count_room_living  \
42615  ...               1.0                 NaN                NaN   
42616  ...               1.0                 1.0                NaN   
42617  ...               NaN                 NaN                NaN   
42618  ...               NaN                 NaN                1.0   
42619  ...               1.0                 1.0                1.0   
42620  ...               NaN                 1.0                NaN   
42621  ...               NaN                 NaN                NaN   
42622  ...               1.0                 NaN                NaN   
42623  ...               NaN                 1.0                NaN   
42624  ...               1.0                 1.0                NaN   

       count_room_maid  count_room_storage  count_room_utility  \
42615              NaN                 NaN                 NaN   
42616              NaN                 1.0                 NaN   
42617              NaN                 NaN                 NaN   
42618              NaN                 NaN                 NaN   
42619              1.0                 1.0                 NaN   
42620              NaN                 NaN                 NaN   
42621              NaN                 NaN                 NaN   
42622              NaN                 NaN                 NaN   
42623              NaN                 NaN                 NaN   
42624              NaN                 NaN                 NaN   

              date_created         date_updated  source  \
42615  2022-02-28 15:30:05 

In [ ]:
data1.shape #ถ x ห เป็นเท่าไหร่

(42625, 27)

## Explore data

In [7]:
data1.describe()

row_number  propertytype_id  area_total_min_wa  area_usable_min  \
count  42625.000000     42625.000000        3747.000000     36999.000000   
mean   21313.000000       938.471859          44.413819       139.286943   
std    12304.921949      4220.823322          21.502884       111.939963   
min        1.000000         1.000000           1.000000        14.000000   
25%    10657.000000         1.000000          23.000000        60.000000   
50%    21313.000000         2.000000          50.000000       125.000000   
75%    31969.000000         3.000000          60.000000       179.000000   
max    42625.000000     20000.000000         335.000000      4350.000000   

          price_min  count_airconditioner  count_elevator  \
count  3.542800e+04           1780.000000             0.0   
mean   5.569699e+06              2.465730             NaN   
std    1.117088e+07              1.512717             NaN   
min    1.000000e+05              1.000000             NaN   
25%    1.950000e+06              1.000000             NaN   
50%    3.100000e+06              2.000000             NaN   
75%    5.290000e+06              3.000000             NaN   
max    5.800000e+08             10.000000             NaN   

       count_floor_internal  count_parking  count_room_bath  count_room_bed  \
count          29709.000000   23520.000000     41144.000000    40658.000000   
mean               1.945572       1.844090         2.286263        2.528137   
std                0.658217       0.759404         1.145901        1.158793   
min                0.000000       1.000000         0.000000        0.000000   
25%                2.000000       1.000000         1.000000        2.000000   
50%                2.000000       2.000000         2.000000        3.000000   
75%                2.000000       2.000000         3.000000        3.000000   
max                9.000000      10.000000        11.000000       10.000000   

       count_room_dinning  count_room_guest  count_room_kitchen  \
count        13670.000000      13044.000000        20454.000000   
mean             1.001902          1.010810            1.018138   
std              0.043572          0.106334            0.134549   
min              1.000000          1.000000            1.000000   
25%              1.000000          1.000000            1.000000   
50%              1.000000          1.000000            1.000000   
75%              1.000000          1.000000            1.000000   
max              2.000000          3.000000            3.000000   

       count_room_living  count_room_maid  count_room_storage  \
count       10026.000000      2068.000000         4108.000000   
mean            1.053860         1.048356            1.054771   
std             0.262145         0.227696            0.269673   
min             1.000000         1.000000            1.000000   
25%             1.000000         1.000000            1.000000   
50%             1.000000         1.000000            1.000000   
75%             1.000000         1.000000            1.000000   
max             5.000000         4.000000            5.000000   

       count_room_utility  
count          699.000000  
mean             1.058655  
std              0.284749  
min              1.000000  
25%              1.000000  
50%              1.000000  
75%              1.000000  
max              4.000000

- NaN = Not a number
- count --> นับจำนวนแถว แต่ที่ไม่เท่ากัน แปลว่า จาก shape มี 42625 แถว แต่ที่บางแถวไม่ครบ เพราะมันไม่นับค่าว่าง(ค่าว่าง = NaN)
- mean --> ค่าเฉลี่ย (44 --> จากบ้านสี่หมื่นกว่าหลังมีพื้นที่เฉลี่ย 44 ตร.วา)
- std --> ค่า Standard Deviation (SD) หรือค่าเบี่ยงเบนมาตรฐาน ค่านี้มีความสำคัญโดยจะใช้บอกค่าเฉลี่ยของระยะห่างของข้อมูลแต่ละตัวจากค่าเฉลี่ย โดยจะเปรียบเทียบกับค่า mean หรือค่าเฉลี่ยร่วมด้วย หากค่า std ต่ำ แปลว่า ค่าของข้อมูลแต่ละตัวไม่ต่างกันมาก หากค่า std สูง แปลว่า ค่าของข้อมูลแต่ละตัวมีความต่างกันมาก มีการกระจายของข้อมูลสูง (กระจายไประหว่าง 44 + - 21 เวลาสุ่มเจอบ่อยมาก)
- min --> ต่ำที่สุด,น้อยสุด (ค่าพื้นที่ที่น้อยที่สุด คือ 1 ตารางวา)
- 25% --> 25% ของบ้านทั้งหมด มีพท.น้อยกว่า 23 ตร.วา (25%ของจำนวนบ้านที่สำรวจมาทั้งหมดสี่หมื่นกว่าหลังมีพื้นที่น้อยกว่า 23ตรวา หรือมีจำนวนพื้นที่มากที่สุดคือ 23 ตรว )
- 50% --> 50% ของบ้านทั้งหมด มีพท.น้อยกว่า 50 ตร.วา
- 75% --> 75% ของบ้านทั้งหมด มีพท.น้อยกว่า 60 ตร.วา (ไม่ต่ำกว่า)
- max --> สูงสุด,มากสุด (ค่าพท.บ้านที่มากที่สุด คือ 335 ตร.วา)
- price  e+08 = 10^8แพงสุด 58 ล้าน
- parking มีมากสุด 10 ที่